### ISSUE TITLE
[Bug] https://github.com/apache/spark/pull/42760

### ISSUE THREAD

**Issue by cs-lpc** [apache/kyuubi#5230]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

According to the PR https://github.com/apache/spark/pull/42760, if the hive.metastore.token.signature parameter is not empty, we may also encounter authentication failures in Kyuubi.

1. Set the configuration bellow
```
    <property>
       <name>hive.metastore.token.signature</name>
       <value>HIVE_DELEGATION_TOKEN</value>
    </property>
```
3. Start kyuubi server with `hive`
4. Connet kyuubi with proxy user ，
5. Run sql `insert into test11 values('a');` 

then the user  will be `hive` to execute hdfs command

### Affects Version(s)

master

### Kyuubi Server Configurations

```yaml
<property>
       <name>hive.metastore.token.signature</name>
       <value>HIVE_DELEGATION_TOKEN</value>
    </property>
```

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

**Comment by cs-lpc** [comment]
Before Spark fixes this bug, Kyuubi should also be compatible with this behavior.

**Comment by pan3793** [comment]
Do you have suggestions? Maybe forcibly overwriting `hive.metastore.token.signature` during `spark-submit`?

**Comment by pan3793** [comment]
cc @zhouyifan279

**Comment by cs-lpc** [comment]
After merging https://github.com/apache/spark/pull/42760, directly overwriting this configuration may cause new exceptions. We need to be provied new configuration. By adjusting the default value of this configuration can  adapt to different versions of Spark.

**Comment by pan3793** [comment]
Yes, the overwriting behavior should be judged by the Spark version and also allowed to be disabled in case some vendor's Spark version backports it into lower Spark versions.

**Comment by cxzl25** [comment]
It sounds like a misconfiguration, `hive.metastore.token.signature` should not be configured in `hive-site.xml`.

**Comment by cs-lpc** [comment]
https://github.com/apache/kyuubi/pull/5261
